tests/unit/unit2604.c

```diff
@@ -104,7 +104,9 @@ UNITTEST_START
   }
 
   free((void *)list[0].cp);
-  return error == 0 ? CURLE_OK : TEST_ERR_FAILURE;
+  if(error == 0) {
+    return CURLE_OK;
+  }
```

Comment:
This looks wrong. What if error is not zero, what does it return then?

Reply:
It going to this line and return:
return (CURLcode)unitfail; 
https://github.com/curl/curl/blob/master/tests/unit/curlcheck.h#L110

Reply:
What do you think should be the value? 
Maybe this line:
https://github.com/curl/curl/blob/master/tests/unit/curlcheck.h#L110
should return TEST_ERR_FAILURE?

Reply:
I think I know what needed to be done. I understand my error.
I think I should call abort_test.